compiler/rustc_data_structures/src/sync.rs

```diff
@@ -26,6 +26,36 @@ use std::panic::{catch_unwind, resume_unwind, AssertUnwindSafe};
 pub use std::sync::atomic::Ordering;
 pub use std::sync::atomic::Ordering::SeqCst;
 
+mod mode {
+    use super::Ordering;
+
+    const UNINITIALIZED: usize = 0;
+    const SINGLE_THREAD: usize = 1;
+    const MULTIPLE_THREADS: usize = 2;
+
+    static PARALLEL: std::sync::atomic::AtomicUsize =
+        std::sync::atomic::AtomicUsize::new(UNINITIALIZED);
```

Comment:
What if a concurrent independent rustc session in the same process enables parallel mode? Or worse sets non-parallel mode?

Reply:
They will panic if they set different modes, so this is an per-executable setting.